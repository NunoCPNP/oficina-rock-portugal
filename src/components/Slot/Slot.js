import { useEffect, useState } from 'react'
import useTranslation from 'next-translate/useTranslation'

import { Container, Section } from './Slot.styles'

const Slot = () => {
  const [item, setItem] = useState(0)
  const { t } = useTranslation('common')

  const content = [t(`slot-0`), t(`slot-1`)]
  const timer = 10000

  useEffect(() => {
    setTimeout(() => {
      setItem(item < content.length - 1 ? item + 1 : 0)
    }, timer)
  }, [content.length, item, timer])

  return (
    <Container>
      <Section>{content[item]}</Section>
    </Container>
  )
}

export default Slot
