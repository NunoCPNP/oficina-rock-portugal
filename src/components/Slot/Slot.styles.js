import styled from '@emotion/styled'

export const Container = styled.div`
  background-color: var(--ghostWhite);
  display: flex;
  justify-content: center;
  text-transform: uppercase;
  font-size: var(--heading-5);
  position: relative;
  z-index: var(--headerLevel);
`

export const Section = styled.div`
  padding: 0.5rem;
`
