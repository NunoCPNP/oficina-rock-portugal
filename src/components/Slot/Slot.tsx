type Props = {}

const Slot: React.FC<Props> = () => {
  return <div></div>
}

export default Slot