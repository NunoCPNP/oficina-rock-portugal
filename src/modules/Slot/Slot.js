const Slot = () => {
  return <div></div>;
};

export default Slot;
