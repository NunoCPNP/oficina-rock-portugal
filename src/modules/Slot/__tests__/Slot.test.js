import { render } from '@testing-library/react'

import Slot from '../Slot'

describe('<Slot /> spec', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Slot />)

    expect(container).toMatchSnapshot()
  })
})
