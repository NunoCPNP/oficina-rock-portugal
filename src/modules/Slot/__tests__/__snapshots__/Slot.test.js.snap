// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slot /> spec Should match snapshot 1`] = `
.emotion-0 {
  background-color: var(--ghostWhite);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-transform: uppercase;
  font-size: 1.4rem;
  position: relative;
  z-index: var(--headerLevel);
}

.emotion-1 {
  padding: 0.5rem;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      common:slot-0
    </div>
  </div>
</div>
`;
